File: ipyhtmx/htmx.py
import ipywidgets as widgets
from traitlets import Unicode
from ._version import NPM_PACKAGE_RANGE

@widgets.register
class Htmx(widgets.DOMWidget):
    """An example widget."""

    # Name of the widget view class in front-end
    _view_name = Unicode('HtmxView').tag(sync=True)

    # Name of the widget model class in front-end
    _model_name = Unicode('HtmxModel').tag(sync=True)

    # Name of the front-end module containing widget view
    _view_module = Unicode('ipyhtmx').tag(sync=True)

    # Name of the front-end module containing widget model
    _model_module = Unicode('ipyhtmx').tag(sync=True)

    # Version of the front-end module containing widget view
    _view_module_version = Unicode(NPM_PACKAGE_RANGE).tag(sync=True)
    # Version of the front-end module containing widget model
    _model_module_version = Unicode(NPM_PACKAGE_RANGE).tag(sync=True)

    html = Unicode().tag(sync=True)

File: js/lib/htmx.js
import { DOMWidgetModel, DOMWidgetView } from '@jupyter-widgets/base';
import htmx from 'htmx.org';

export class HtmxModel extends DOMWidgetModel {
    defaults() {
      return {
        ...super.defaults(),
        _model_name : 'HtmxModel',
        _view_name : 'HtmxView',
        _model_module : 'ipyhtmx',
        _view_module : 'ipyhtmx',
        _model_module_version : '0.1.0',
        _view_module_version : '0.1.0',
        html : ''
      };
    }
  }

export class HtmxView extends DOMWidgetView {
    render() {
        this.value_changed();

        this.model.on('change:html', this.value_changed, this);
    }

    value_changed() {
        this.el.innerHTML = this.model.get('html');
        htmx.process(this.el);
    }
}
